src/api.rs

```diff
@@ -141,10 +141,13 @@ async fn add_file(
     hash: extract::Path<String>,
     file: Upload,
 ) -> Result<OutpackSuccess<()>, OutpackError> {
-    store::put_file(&root, file, &hash)
-        .await
-        .map_err(OutpackError::from)
-        .map(OutpackSuccess::from)
+    tokio::task::spawn_blocking(move || {
+        store::put_file(&root, file, &hash)
+            .map_err(OutpackError::from)
+            .map(OutpackSuccess::from)
+    })
+    .await
+    .unwrap()
```

Comment:
so this just moves it to another thread?

Reply:
Yeah. Because the `store` module's API isn't async we need to be careful with calling functions that do heavy IO on it. I didn't wrap the more metadata-y ones on the basis that they probably won't block for that long.

The alternative would be to write everything async style and use `tokio::fs` everywhere instead of `std::fs`.